### PR TITLE
refactor(Banner): update region to use a dedicated aria-label

### PR DESCRIPTION
**Pull request by joshblack**
Closes https://github.com/github/primer/issues/3262

#### Changed

- Update Banner based on the changes to the API doc so that it is no longer `aria-labelledby` the heading and is instead `aria-label` with the variant info

### Rollout strategy

- [x] Minor release

### Testing & Reviewing

- Verify that the change reflects the API docs changes
- Make sure that test scenarios have been accurately updated to reflect the changes
- Is there anything I'm missing in this implementation? 👀

### CODE REVIEW

File: packages/react/src/Banner/Banner.test.tsx [hunk 1]
```diff
@@ -25,12 +25,38 @@ describe('Banner', () => {
 
   it('should render as a region element', () => {
     render(<Banner title="test" />)
-    expect(screen.getByRole('region', {name: 'test'})).toBeInTheDocument()
+    expect(screen.getByRole('region', {name: 'Information'})).toBeInTheDocument()
+    expect(screen.getByRole('heading', {name: 'test'})).toBeInTheDocument()
   })
 
-  it('should label the landmark element with the title prop', () => {
+  it('should label the landmark element with the corresponding variant label text', () => {
```
[the remark this answers is not in the document]
Comment (joshblack) [reply packages/react/src/Banner/Banner.test.tsx:32]: @khiga8 love this suggestion, let me add it real quick!

File: packages/react/src/Banner/Banner.test.tsx [hunk 2]
```diff
@@ -25,14 +25,45 @@ describe('Banner', () => {
...
+    expect(screen.getByRole('region')).toEqual(screen.getByLabelText('Recommendation'))
+  })
+
+  it('should label the landmark element with the label for the warning variant', () => {
+    render(<Banner title="test" variant="warning" />)
+    expect(screen.getByRole('region')).toEqual(screen.getByLabelText('Warning'))
+  })
+
+  it('should support the `aria-label` prop to override the default label for the landmark', () => {
+    render(<Banner aria-label="test" title="test" variant="warning" />)
```
Comment (khiga8) [inline packages/react/src/Banner/Banner.test.tsx:63]:
```suggestion
    render(<Banner aria-label="Test" title="test" variant="warning" />)
```

File: packages/react/src/Banner/Banner.test.tsx [hunk 3]
```diff
@@ -25,14 +25,45 @@ describe('Banner', () => {
...
+    expect(screen.getByRole('region')).toHaveAttribute('aria-label', 'test')
+  });
```
Comment (khiga8) [inline packages/react/src/Banner/Banner.test.tsx:65]:
```suggestion
  })
```

**Review by khiga8 — APPROVED** [review]
🎉

File: packages/react/src/Banner/Banner.tsx [hunk 4]
```diff
@@ -64,74 +71,96 @@ const iconForVariant: Record<BannerVariant, React.ReactNode> = {
...
+      const {current: banner} = bannerRef
+      if (!banner) {
+        return
+      }
+
+      const hasTitle = banner.querySelector('[data-banner-title]')
+      if (!hasTitle) {
         throw new Error(
-          'The Banner component requires a title to be provided as the `title` prop or through `Banner.Title`',
+          'Expected a title to be provided to the <Banner> component with the `title` prop or through `<Banner.Title>` but no title was found',
```
Comment (broccolinisoup) [inline packages/react/src/Banner/Banner.tsx:119]: What is a good way to differentiate `Error` vs `console.error (or invariant)` in these type of scenarios like when to use one over the other one? 👀

**Review by broccolinisoup — APPROVED** [review]
Just left one question 😊 Looks good!

File: packages/react/src/Banner/Banner.tsx (hunk 4 above)
Comment (joshblack) [reply packages/react/src/Banner/Banner.tsx:119]: @broccolinisoup great question honestly, they kind of each have a niche and I'm not sure when it's best to use which 🤔 Would this be a good topic for a working session?

Comment (broccolinisoup) [reply packages/react/src/Banner/Banner.tsx:119]: I think it is a great idea 🔥 I had related questions in mind as well. I'll add it to the agenda!

**Pull request merged**